### ISSUE TITLE
Deleting a folder

### ISSUE THREAD

**Issue by Zungaron** [iterative/PyDrive2#117]
Unable to delete a folder when calling it by id number and in subroutines other than creation scope.
Code using ID:
```
folder = googleDrive.CreateFile({'mimeType':'application/vnd.google-apps.folder', 'id':'1ZPWs_kA6b-Y2dzOWYr-8lZvif378wih1'})
folder.Delete()
```
I am able to upload files to the folder with the ID numbers. The ID numbers themselves are functional.
`Message=<HttpError 404 when requesting https://www.googleapis.com/drive/v2/files/1ZPWs_kA6b-Y2dzOWYr-8lZvif378wih1? returned "File not found: 1ZPWs_kA6b-Y2dzOWYr-8lZvif378wih1". Details: "[{'domain': 'global', 'reason': 'notFound', 'message': 'File not found: 1ZPWs_kA6b-Y2dzOWYr-8lZvif378wih1', 'locationType': 'other', 'location': 'file'}]">`

I can .delete() a folder in **only one** condition: within the subroutine where I create the folder and .upload() it, and only using the folder name variable / string, not ID. eg.
`.CreateFile({'title':folderName, 'mimeType':'application/vnd.google-apps.folder'})`
In any other subroutine I get a blank exception when I try to CreateFile an object to .Delete(). The only thing that happens in the meantime is a couple files being uploaded to the created folder. I ommited uploading the files and code still failed at the .delete() call, with a blank exception.

**Comment by Zungaron** [comment]
I can execute .delete() on a folder in other subroutines if I initially create the folder as a global variable (for testing)
eg.
`global folder
folder = googleDrive.CreateFile({'title':folderName, 'mimeType':'application/vnd.google-apps.folder'})
folder.Upload()`

I can **not** delete a folder in another subroutine when I try to reference it by ID through .CreateFile()
eg.
`folder = googleDrive.CreateFile({'mimeType':'application/vnd.google-apps.folder', 'id':'1ZPWs_kA6b-Y2dzOWYr-8lZvif378wih1'})
folder.Delete()`

However, this is how the documentation describes using .Delete(). Please clarify how I can delete an existing folder.

**Comment by Zungaron** [comment]
My problem may have been due to some small error on my part. I am not sure what fixed it.

I changed my capture of the folder ID from
`folderID = googleDrive.ListFile({'q': "title = '"+folderName+"' and trashed=false"}).GetList()[0]`
to
`folderID = folder['id']`
I was able to upload files to the ID created with the first method, however. It seemed functional.

I also changed how I retreive the ID number from a text file, from 
`with open ("FolderIDs.txt", "rt") as file:`
to 
`with open ("FolderIDs.txt") as file:`
read and text modes are default for open, however. This should not have changed anything.

Maybe I fixed some small typo. I was not seeing any undefined variables, and the only error I was getting was a blank exception.
Thanks to Ivan for his help!
